Fix periodic snapshot task and replication task documentation inaccur… (#1299)
freenas/freenas-docs: userguide/storage.rst

   |                   |                   |                                                                                            |
   +-------------------+-------------------+--------------------------------------------------------------------------------------------+
   | Exclude           | string            | Exclude specific child datasets from the snapshot. Use with :guilabel:`Recursive`          |
   |                   |                   | snapshots. List names of child datasets to exclude, one per line.                          |
   |                   |                   | Example: :samp:`pool1/dataset1/child1`. A recursive snapshot of :file:`pool1/dataset1`     |

>    |                   |                   | includes all child datasets except :file:`child1` and it's descendants.                    |

Author: Do we (or anybody other) say "dataset tank/data is below dataset tank"?

>    |                   |                   | includes all child datasets except :file:`child1` and it's descendants.                    |

Reviewer: How about "includes all child datasets except child1 and any datasets inside it."?

:ref:`Periodic Snapshot Tasks` to generate local copies of important
data and replicate these copies to a remote system.

Replications require a source system with dataset snapshots and a
destination that can store the copied data. Remote replications require

> a saved :ref:`SSH Connection <SSH Connections>` on the local system and

Author: Problem is that we now have push replication, so "local = source" and "remote = destination" does not work anymore, that's why I am suggesting this.

:ref:`Periodic Snapshot Tasks` to generate local copies of important
data and replicate these copies to a remote system.

Replications require a source system with dataset snapshots and a
destination that can store the copied data. Remote replications require
a saved :ref:`SSH Connection <SSH Connections>` on the source system and
the destination system must be configured to allow :ref:`SSH`

> connections. Local replications do not require configuring SSH.

Reviewer: Clarify "require configuring":
```suggestion
connections. Local replications do not use SSH.
```

are all possible with this screen.

The :guilabel:`Transport` value changes many of the options for

> replication. :numref:`Table %s <zfs_add_replication_task_opts_tab>` uses

Reviewer: ```suggestion
replication. :numref:`Table %s <zfs_add_replication_task_opts_tab>` shows
```

> shortened versions of the :guilabel:`Transport` methods in the

Reviewer: ```suggestion
abbreviated names of the transport methods in the
```

> :literal:`Transport` column to show which fields appear with the

Reviewer: ```suggestion
:literal:`Transport` column to identify fields which appear when that method is selected.
```

   +===========================+===========+================+=================================================================================================================+

>    | Name                      | ALL       | string         | Enter a descriptive :guilabel:`Name` for the replication.                                                       |

Reviewer: Use upper/lower "All" to differentiate the non-literal "All" from the specific types in upper case.

   |                           |           |                | * *LEGACY* uses the legacy replication engine from %brand% 11.2 and earlier.                                    |
   |                           |           |                |                                                                                                                 |
   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+
   | SSH Connection            | SSH, NCT, | drop-down menu | Choose the :ref:`SSH connection <SSH Connections>`.                                                             |
   |                           | LEG       |                |                                                                                                                 |
   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+

>    | Netcat Active Side        | NCT       | drop-down menu | Choose the system (*LOCAL* or *REMOTE*) that will open TCP ports to configure the connection between the two    |

Author: To create a TCP connection one side (server) needs to open TCP port to which other side (client) can connect to.

>    | Netcat Active Side        | NCT       | drop-down menu | Choose the system (*LOCAL* or *REMOTE*) that will open TCP ports to configure the connection between the two    |

Reviewer: Okay, so it's really "Choose the system that will open receiving ports." Not clear how that is the "active" side, maybe call it the "listening side"?

   | Target Dataset            | ALL       | string         | Enter the dataset name on the destination system where snapshots are stored. Example:                           |
   |                           |           |                | :samp:`{pool1}/{dataset1}`, where *pool1* is the name of the top-level storage pool dataset and *dataset1* is   |
   |                           |           |                | the name of the dataset to store replicated snapshots.                                                          |
   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+
   | Recursive                 | ALL       | checkbox       | Replicate all child dataset snapshots. Set to make :guilabel:`Exclude Child Datasets` visible.                  |
   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+

>    | Exclude Child Datasets    | SSH, NCT, | string         | Exclude specific child dataset snapshots from the replication. Use with :guilabel:`Recursive` snapshots. List   |

Reviewer: ```suggestion
   | Exclude Child Datasets    | SSH, NCT, | string         | Exclude specific child datasets from the replication. Use with :guilabel:`Recursive` snapshots. List   |
```

   |                           | LOC       |                | child dataset names to exclude. Example: :samp:`pool1/dataset1/child1`. A recursive replication of              |

>    |                           |           |                | :file:`pool1/dataset1` snapshots includes all child dataset snapshots except :file:`child1`.                    |

Reviewer: ```suggestion
   |                           |           |                | :file:`pool1/dataset1` snapshots includes all child datasets except :file:`child1`.                    |
```

   |                           |           |                | periodic snapshot task takes a snapshot every hour, but only every other snapshot is needed for replication.    |
   |                           |           |                | The scheduler is set to even hours and only snapshots taken at those times are replicated.                      |
   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+

>    | Begin                     | SSH, NCT, | drop-down menu | Set a starting time for the replication task to run.                                                            |

Reviewer: ```suggestion
   | Begin                     | SSH, NCT, | drop-down menu | Start time for the replication task.                                                            |
```

   |                           | LOC       |                |                                                                                                                 |
   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+

>    | End                       | SSH, NCT, | drop-down menu | Set an ending time for the replication task. A replication that is in progress can continue to run past this    |

Reviewer: ```suggestion
   | End                       | SSH, NCT, | drop-down menu | End time for the replication task. A replication that is already running can continue past this    |
```

   | End                       | SSH, NCT, | drop-down menu | Define an ending time for the replication task. A replication that is in progress will continue to              |
   |                           | LOC       |                | run past this time.                                                                                             |
   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+
   | Only Replicate Snapshots  | SSH, NCT, | checkbox       | Set to either use the :guilabel:`Schedule` in place of the :guilabel:`Snapshot Replication Schedule` or add     |
   | Matching Schedule         | LOC       |                | the :guilabel:`Schedule` values to the :guilabel:`Snapshot Replication Schedule`.                               |
   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+

>    | Replicate from scratch if | SSH, NCT, | checkbox       | If this is enabled and destination system does have snapshots but none of them matches any snapshot on source   |

Author: I like "Perform full replication if incremental is impossible" and our UI can tolerate this perfectly. Can't think of anything that can be significantly shorter and still explain a purpose of this clearly. Do you think we should make this change?

   | Number of retries for     | SSH, NCT, | integer        | Number of times the replication is attempted before stopping and marking the task as failed.                    |
   | failed replications       | LOC       |                |                                                                                                                 |
   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+

>    | Logging Level             | ALL       | drop-down menu | Verbosity of the messages in replication task log.                                                              |

Reviewer: ```suggestion
   | Logging Level             | ALL       | drop-down menu | Verbosity level of messages in the replication task log.                                                          |
```

task. There are options to :guilabel:`Delete`, :guilabel:`Edit`, or
immediately start that replication.

> Replications will run in parallel (as long as they don't conflict with each other).

Reviewer: Use present tense, avoid asides, avoid contractions.
```suggestion
Replications run in parallel as long as they do not conflict with each other.
```

encryption to work. A failure or misconfiguration of any of these can
prevent successful replication.

> All replication logs are located in :file:`/var/log/zettarepl.log`. You can

Reviewer: Avoid "you" or "your".
```suggestion
Replication logs are saved in :file:`/var/log/zettarepl.log` and can be
```

Reviewer: Is it too late to change the name of the log file to a generic `replication.log`?  Using the name of a particular implementation in the log filename will often become a hassle down the road, like if the replication system changes.  replication.log would work for anything.

Author: User won't need to interact with this file, we have logs in UI that are split per-task. I've named it after zettarepl because that's how other files in /var/log are named: nginx/access.log, not webserver/access.log; proftpd.log, not ftp.log, etc. If the replication system changes, replication.log will change it's format too and being on the same place won't give nobody any help.

> inspect logs of any specific replication task by clicking on it's state.

Reviewer: As above about possessives: no apostrophe in possessive "its", but avoid possessives anyway because they are less clear.

Finally, it's not clear what "state" is referring to here.  Probably should use at least `` :guilabel:`State` `` if it is a UI element.

```suggestion
inspected clicking on their state.
```
